File: backend/schemas/player.py
from pydantic import BaseModel, EmailStr, Field
from typing import Optional, Literal
from uuid import UUID
from datetime import datetime


class PlayerBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    nickname: Optional[str] = Field(None, max_length=50)
    email: EmailStr
    phone: Optional[str] = Field(None, max_length=20)
    skill_level: int = Field(default=0, ge=0, le=3)
    gender: Optional[Literal['M', 'F']] = None


class PlayerCreate(PlayerBase):
    password: str = Field(..., min_length=8, max_length=100)


class PlayerUpdate(BaseModel):
    name: Optional[str] = Field(None, min_length=1, max_length=100)
    nickname: Optional[str] = Field(None, max_length=50)
    email: Optional[EmailStr] = None
    phone: Optional[str] = Field(None, max_length=20)
    skill_level: Optional[int] = Field(None, ge=0, le=3)
    is_active: Optional[bool] = None
    gender: Optional[Literal['M', 'F']] = None


class PlayerResponse(PlayerBase):
    id: UUID
    is_active: bool
    marketing_opt_in: bool = False
    qr_code: Optional[str]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class PlayerLogin(BaseModel):
    email: EmailStr
    password: str


class PlayerSelfRegister(BaseModel):
    """Schema for public player self-registration (no password required)."""
    name: str = Field(..., min_length=1, max_length=100)
    email: EmailStr
    phone: str = Field(..., min_length=1, max_length=20)
    nickname: Optional[str] = Field(None, max_length=50)
    gender: Optional[Literal['M', 'F']] = None

File: shared/types/player.ts
export interface Player {
  id: string;
  name: string;
  email: string;
  phone?: string;
  skill_level: number;
  is_active: boolean;
  qr_code?: string;
  gender?: 'M' | 'F';
  created_at: string;
  updated_at: string;
}

export interface PlayerCreate {
  name: string;
  email: string;
  password: string;
  phone?: string;
  skill_level?: number;
  gender?: 'M' | 'F';
}

export interface PlayerUpdate {
  name?: string;
  email?: string;
  phone?: string;
  skill_level?: number;
  is_active?: boolean;
  gender?: 'M' | 'F';
}

export interface PlayerLogin {
  email: string;
  password: string;
}

export interface PlayerStats {
  average: number;
  total_darts: number;
  highest_score: number;
  doubles_hit: number;
  triples_hit: number;
  total_score: number;
}
